ledger-tool/src/accounts.rs

```diff
@@ -0,0 +1,175 @@
+//! The `accounts_tool` subcommand
+use {
+    crate::utils::*,
+    clap::{value_t, App, AppSettings, Arg, ArgMatches, SubCommand},
+    log::*,
+    serde::ser::{SerializeSeq, Serializer},
+    solana_cli_output::{CliAccount, CliAccountNewConfig, OutputFormat},
+    solana_ledger::{
+        blockstore_options::{AccessType, BlockstoreRecoveryMode},
+        blockstore_processor::ProcessOptions,
+    },
+    solana_measure::measure::Measure,
+    solana_runtime::{accounts::Accounts, bank::TotalAccountsStats},
+    solana_sdk::{
+        account::{AccountSharedData, ReadableAccount},
+        clock::Slot,
+        pubkey::Pubkey,
+    },
+    std::{
+        io::stdout,
+        path::{Path, PathBuf},
+        process::exit,
+        sync::Arc,
+    },
+};
+
+pub trait AccountsSubCommand<'a> {
+    fn accounts_subcommand(
+        self,
+        no_snapshot_arg: &'a Arg,
+        account_paths_arg: &'a Arg,
+        halt_at_slot_arg: &'a Arg,
+        hard_forks_arg: &'a Arg,
+        geyser_plugin_args: &'a Arg,
+        accounts_data_encoding_arg: &'a Arg,
+        max_genesis_archive_unpacked_size_arg: &'a Arg,
+    ) -> Self;
```

Comment:
Hmm, this is pretty annoying but don't think I see a way around it

Reply:
Actually, remind me, the new commands you're planning to add wouldn't be unpacking snapshot / creating a bank right ? If so, maybe we could only put the new subcommand into the new file?

Let me think on this and/or let's chat before you do anything ... I know we've already gone back and forth several times on what direction so I would hate for you incur that again

Reply:
> Actually, remind me, the new commands you're planning to add wouldn't be unpacking snapshot / creating a bank right ? If so, maybe we could only put the new subcommand into the new file?

You're right.  The new subcommands so far don't use snapshot and blockstore.  Our previous consensus is that we want to have the new commands under "accounts" as they are accounts-related subcommands.

So the default behavior of "accounts" will still be the same as the existing one.  But if you run "accounts" with additional subcommands like "accounts top", then it will run the new command.